conda_lock/pypi_solver.py

```diff
@@ -327,9 +327,11 @@ def get_dependency(dep: lock_spec.Dependency) -> PoetryDependency:
             hash=dep.hash,
         )
     elif isinstance(dep, lock_spec.URLDependency):
+        hash_string = f"#{dep.hashes[0]}" if len(dep.hashes[0]) > 0 else ""
+
         return PoetryURLDependency(
             name=dep.name,
-            url=f"{dep.url}#{dep.hashes[0].replace(':','=')}",
+            url=f"{dep.url}" + hash_string.replace(':','='),
```

Comment:
thanks Dom, I wonder if the fix will be enough with non-empty hash in pyproject.toml

We are having these when pointing to a git tag.
Maybe we should not care too much: only a safety measure in case a tag is moving.

Good for now. We will see